tests/test_odilo.py

```diff
@@ -0,0 +1,342 @@
+# encoding: utf-8
+import json
+
+from nose.tools import (
+    eq_, ok_, assert_raises,
+)
+
+from api.odilo import (
+    MockOdiloAPI,
+    RecentOdiloCollectionMonitor,
+    FullOdiloCollectionMonitor
+)
+
+from api.circulation import (
+    CirculationAPI,
+)
+
+from api.circulation_exceptions import *
+
+from . import (
+    DatabaseTest,
+    sample_data
+)
+
+from core.model import (
+    DataSource,
+    ExternalIntegration,
+    Identifier,
+)
+
+
+class OdiloAPITest(DatabaseTest):
+    PATRON = '0001000265'
+    PIN = 'c4ca4238a0b923820dcc509a6f75849b'
+    RECORD_ID = '00010982'
+
+    def setup(self):
+        super(OdiloAPITest, self).setup()
+        library = self._default_library
+        self.collection = MockOdiloAPI.mock_collection(self._db)
+        self.circulation = CirculationAPI(
+            self._db, library, api_map={ExternalIntegration.ODILO: MockOdiloAPI}
+        )
+        self.api = self.circulation.api_for_collection[self.collection.id]
+
+        self.edition, self.licensepool = self._edition(
+            data_source_name=DataSource.ODILO,
+            identifier_type=Identifier.ODILO_ID,
+            collection=self.collection,
+            identifier_id=self.RECORD_ID,
+            with_license_pool=True
+        )
+
+    @classmethod
+    def sample_data(cls, filename):
+        return sample_data(filename, 'odilo')
+
+    @classmethod
+    def sample_json(cls, filename):
+        data = cls.sample_data(filename)
+        return data, json.loads(data)
+
+    def error_message(self, error_code, message=None, token=None):
+        """Create a JSON document that simulates the message served by
+        Odilo given a certain error condition.
+        """
+        message = message or self._str
+        token = token or self._str
+        data = dict(errorCode=error_code, message=message, token=token)
+        return json.dumps(data)
+
+
+class TestOdiloCirculationAPI(OdiloAPITest):
+    #################
+    # General tests
+    #################
+
+    # Test 404 Not Found --> patron not found --> 'patronNotFound'
+    def test_01_patron_not_found(self):
```

Comment:
It shouldn't be necessary to number the test cases.

Reply:
Yes, I know, and it isn't "necessary", it's just "useful" for me to have them numbered, in order to ensure myself that I have tested all things.
The tests are independent among them and could be run in any order.